generator/transit_generator.cpp

```diff
@@ -321,7 +245,196 @@ void BuildTransit(string const & mwmDir, string const & countryId,
+
+  for (auto & e : m_edges)
+  {
+    if (e.GetWeight() != kInvalidWeight)
+      continue;
+
+    Stop const & s1 = FindStopById(m_stops, e.GetStop1Id());
+    Stop const & s2 = FindStopById(m_stops, e.GetStop2Id());
+    double const lengthInMeters = MercatorBounds::DistanceOnEarth(s1.GetPoint(), s2.GetPoint());
+    e.SetWeight(lengthInMeters / kTransitAverageSpeedMPS);
+  }
+}
+
+void GraphData::CalculateBestPedestrianSegments(string const & mwmPath, string const & countryId)
+{
+  // Creating IndexRouter.
+  SingleMwmIndex index(mwmPath);
+
+  auto infoGetter = storage::CountryInfoReader::CreateCountryInfoReader(GetPlatform());
+  CHECK(infoGetter, ());
+
+  auto const countryFileGetter = [&infoGetter](m2::PointD const & pt) {
+    return infoGetter->GetRegionCountryId(pt);
+  };
+
+  auto const getMwmRectByName = [&](string const & c) -> m2::RectD {
+    CHECK_EQUAL(countryId, c, ());
+    return infoGetter->GetLimitRectForLeaf(c);
+  };
+
+  CHECK_EQUAL(index.GetMwmId().GetInfo()->GetType(), MwmInfo::COUNTRY, ());
+  auto numMwmIds = make_shared<NumMwmIds>();
+  numMwmIds->RegisterFile(CountryFile(countryId));
+
+  // Note. |indexRouter| is valid while |index| is valid.
+  IndexRouter indexRouter(VehicleType::Pedestrian, false /* load altitudes */,
+                          CountryParentNameGetterFn(), countryFileGetter, getMwmRectByName,
+                          numMwmIds, MakeNumMwmTree(*numMwmIds, *infoGetter),
+                          traffic::TrafficCache(), index.GetIndex());
+
+  // Looking for the best segment for every gate.
+  for (auto & gate : m_gates)
+  {
+    // Note. For pedestrian routing all the segments are considered as twoway segments so
+    // IndexRouter.FindBestSegment() method finds the same segment for |isOutgoing| == true
+    // and |isOutgoing| == false.
+    Segment bestSegment;
+    try
+    {
+       if (countryFileGetter(gate.GetPoint()) != countryId)
+         continue;
+      // @todo(bykoianko) For every call of the method below WorldGraph is created. It's not
+      // efficient to create WorldGraph for every gate. The code should be redesigned.
+      if (indexRouter.FindBestSegmentInSingleMwm(gate.GetPoint(),
+                                                 m2::PointD::Zero() /* direction */,
+                                                 true /* isOutgoing */, bestSegment))
+      {
+        CHECK_EQUAL(bestSegment.GetMwmId(), 0, ());
+        gate.SetBestPedestrianSegment(SingleMwmSegment(
+            bestSegment.GetFeatureId(), bestSegment.GetSegmentIdx(), bestSegment.IsForward()));
+      }
+    }
+    catch (MwmIsNotAliveException const & e)
+    {
+      LOG(LCRITICAL, ("Point of a gate belongs to the processed mwm:", countryId, ","
+          "but the mwm is not alive. Gate:", gate, e.what()));
+    }
+    catch (RootException const & e)
+    {
+      LOG(LCRITICAL, ("Exception while looking for the best segment here gates. CountryId::",
```

Comment:
"best segment here gates"? What does it mean?

Reply:
`Exception while looking for the best segment of a gate.`